http3/client.go

```diff
@@ -123,9 +123,9 @@ func (c *client) dial(ctx context.Context) error {
 	}()
 
 	if c.opts.StreamHijacker != nil {
-		go c.handleBidirectionalStreams()
+		go c.handleBidirectionalStreams(ctx)
 	}
-	go c.handleUnidirectionalStreams()
+	go c.handleUnidirectionalStreams(ctx)
```

Comment:
That's not correct. We can't stop unidirectional stream handling when the request is canceled.